examples/ml-pipelines-node-classification.ipynb

```diff
@@ -0,0 +1,1540 @@
+       "      <td>100</td>\n",
+       "      <td>[0.002836007543472271, 0.004250369599202609, 0...</td>\n",
+       "      <td>89.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>8</th>\n",
+       "      <td>8</td>\n",
+       "      <td>100</td>\n",
+       "      <td>[0.018410346867102398, 0.02619614116987539, 0....</td>\n",
+       "      <td>54.0</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9</th>\n",
+       "      <td>9</td>\n",
+       "      <td>4</td>\n",
+       "      <td>[0.011515584874249689, 0.33174804059304086, 0....</td>\n",
+       "      <td>46.0</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   nodeId  predictedClass                             predictedProbabilities  \\\n",
+       "0       0             100  [0.013618774341613382, 0.0008177054013954473, ...   \n",
+       "1       1               1  [0.3186482744624026, 0.02609876129717121, 0.12...   \n",
+       "2       2               2  [0.007710767899246522, 0.7663149051403245, 0.0...   \n",
+       "3       3               2  [0.00350003910223565, 0.9768116873834239, 0.00...   \n",
+       "4       4               3  [0.03084788108403201, 0.0013559744680570265, 0...   \n",
+       "5       5               1  [0.2750609369819228, 0.26605086734276934, 0.09...   \n",
+       "6       6               6  [0.01777377696320589, 0.06070786119694413, 0.0...   \n",
+       "7       7             100  [0.002836007543472271, 0.004250369599202609, 0...   \n",
+       "8       8             100  [0.018410346867102398, 0.02619614116987539, 0....   \n",
+       "9       9               4  [0.011515584874249689, 0.33174804059304086, 0....   \n",
+       "\n",
+       "   confidence  \n",
+       "0        73.0  \n",
+       "1        31.0  \n",
+       "2        76.0  \n",
+       "3        97.0  \n",
+       "4        86.0  \n",
+       "5        27.0  \n",
+       "6        82.0  \n",
+       "7        89.0  \n",
+       "8        54.0  \n",
+       "9        46.0  "
+      ]
+     },
+     "execution_count": 113,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "predicted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "rL7Mi1AIc7OJ"
+   },
+   "source": [
+    "## Adding a data preprocessing step\n",
+    "\n",
+    "The performance of the model can potentially be increased by adding more features or by using different features altogether. One way is to use models such as FastRP that create embeddings based on both node properties and graph features, which can be added via the `addNodeProperty` pipeline method.\n",
+    "\n",
+    "In this example we also use the `contextNodeLabels` parameter to restrict the type of nodes to calculate the embeddings on. This is useful in the prediction phase as it makes unnecessary to explicitly include the labelled nodes.\n",
```

Comment:
I don't understand why to not include the unclassified nodes in the FastRP computation. Well, I guess it depends on what kind of usecase one wants to emulate. Perhaps that should be stated more clearly (unless I missed it)

Reply:
@adamnsch Do you mean to add the unclassified nodes as context as well? Would that be correct?

Reply:
Yes! Well, it depends on what use case you want to emulate. There's no data leakage since you're not using any label information.
You are emulating a user training a model on their data, then getting new data (which is fine btw). But you could also do he case where a user just has data which is not completely labeled, which you mean that you could (and should) use the unlabeled nodes as context

Reply:
In the current example, the holdout nodes are still inserted in the graph but without their `subject` (and with a different label for easier identification, but it could easily be the same label - or no label at all). This looks closer to the second example you've given, since the prediction then is used to add the `subject` to those. Does this mean that "everything" (in this case) should be in the context?

Reply:
The "houldout nodes" would certainly be useful when computing the embeddings as they make the graph more complete :) But what I mean is that if you don't use them, it would be like they didn't exist until the prediction stage (so they're kind of brand new data)